### PR TITLE
Rework structlog, stdlib logging and Sentry

### PR DESCRIPTION
**Pull request by touilleMan**

### CODE REVIEW

File: parsec/core/fs/userfs/userfs.py [hunk 1]
```diff
@@ -1057,11 +1057,7 @@ async def workspace_start_reencryption(self, workspace_id: EntryID) -> Reencrypt
                 workspace_entry.id, new_workspace_entry.encryption_revision, now, reencryption_msgs
             )
             if not ok:
-                # Participant list has changed concurrently
-                logger.info(
```
Comment (AntoineDupre) [inline parsec/core/fs/userfs/userfs.py:1061]: was this log unnecessary ?

File: parsec/cli_utils.py [hunk 2]
```diff
@@ -116,3 +123,77 @@ async def aconfirm(*args, **kwargs):
...
+
+def debug_config_options(fn):
+    decorator = click.option(
+        "--debug",
+        is_flag=True,
+        # Don't prefix with `PARSEC_` given devs are lazy
+        envvar="DEBUG",
+    )
+    decorator(fn)
+    return fn
```
Comment (vxgmichel) [inline parsec/cli_utils.py:199]: `return decorator(fn)`?

**Review by vxgmichel — APPROVED** [review]
LGTM :+1:

File: tests/test_logging.py [hunk 3]
```diff
@@ -0,0 +1,599 @@
...
+        yield events
+
+
+def remove_colors(log: str) -> str:
+    # Much more readable to compare log without all those escapes sequences
+    return re.sub(r"\x1B\[([0-9]{1,2}(;[0-9]{1,2})?)?[mGK]", "", log)
+
+
+def pin_timestamp(log: str) -> str:
+    # Let's fix the future once and for all Marty !
```
Comment (AntoineDupre) [inline tests/test_logging.py:40]: c'est vous le doc doc

**Review by AntoineDupre — APPROVED** [review]
:+1:

File: parsec/cli_utils.py (hunk 2 above)
Comment (touilleMan) [reply parsec/cli_utils.py]: see 5758256

File: parsec/core/fs/userfs/userfs.py (hunk 1 above)
Comment (touilleMan) [reply parsec/core/fs/userfs/userfs.py:1061]: yup, wasn't worth mentioning to the user compared all the concurrency handling that is done in workspace fs transaction module

image

**Pull request merged**